engine/src/main/java/org/terasology/rendering/dag/nodes/ShadowMapNode.java

```diff
@@ -56,50 +56,43 @@
  * - https://docs.google.com/drawings/d/13I0GM9jDFlZv1vNrUPlQuBbaF86RPRNpVfn5q8Wj2lc/edit?usp=sharing
  */
 public class ShadowMapNode extends ConditionDependentNode {
-    public static final ResourceUrn SHADOW_MAP = new ResourceUrn("engine:sceneShadowMap");
+    public static final ResourceUrn SHADOW_MAP_FBO = new ResourceUrn("engine:sceneShadowMap");
+    public static final ResourceUrn SHADOW_MAP_MATERIAL = new ResourceUrn("engine:prog.shadowMap");
     private static final int SHADOW_FRUSTUM_BOUNDS = 500;
     private static final float STEP_SIZE = 50f;
     public Camera shadowMapCamera = new OrthographicCamera(-SHADOW_FRUSTUM_BOUNDS, SHADOW_FRUSTUM_BOUNDS, SHADOW_FRUSTUM_BOUNDS, -SHADOW_FRUSTUM_BOUNDS);
 
-    @In
-    private RenderableWorld renderableWorld;
-
-    @In
-    private RenderQueuesHelper renderQueues;
-
-    @In
-    private Config config;
-
-    @In
     private WorldRenderer worldRenderer;
-
-    @In
+    private RenderQueuesHelper renderQueues;
     private BackdropProvider backdropProvider;
 
-    @In
-    private ShadowMapResolutionDependentFBOs shadowMapResolutionDependentFBOs;
-
     private RenderingConfig renderingConfig;
     private Camera playerCamera;
     private float texelSize;
 
-    @Override
-    public void initialise() {
+    public ShadowMapNode(Context context) {
+        worldRenderer = context.get(WorldRenderer.class);
+        renderQueues = context.get(RenderQueuesHelper.class);
+        backdropProvider = context.get(BackdropProvider.class);
+        Config config = context.get(Config.class);
+        RenderableWorld renderableWorld = context.get(RenderableWorld.class);
+        ShadowMapResolutionDependentFBOs shadowMapResolutionDependentFBOs = context.get(ShadowMapResolutionDependentFBOs.class);
```

Comment:
Let's reorganize the section above so that there is the least distance between the line where the variables are first assigned and where the variables are used. I.e. the worldRenderer line can be followed by the playerCamera assignment.